### PR TITLE
Adds configuration consistency tests

### PR DESCRIPTION
**Pull request by Cyberboss**
Re-PR of #51853

Adds tests to ensure all and only relevant configuration options are "documented" in the sample configs. Added the ones that were missing, removed the ones that were removed.

### CODE REVIEW

File: code/controllers/configuration/configuration.dm [hunk 1]
```diff
@@ -43,11 +43,11 @@
 		CRASH("/datum/controller/configuration/Load() called more than once!")
 	InitEntries()
 	LoadModes()
-	if(fexists("[directory]/config.txt") && LoadEntries("config.txt") <= 1)
+	if(fexists("[directory]/[DEFAULT_CONFIGURATION_FILE]") && LoadEntries(DEFAULT_CONFIGURATION_FILE) <= 1)
 		var/list/legacy_configs = list("game_options.txt", "dbconfig.txt", "comms.txt")
 		for(var/I in legacy_configs)
 			if(fexists("[directory]/[I]"))
-				log_config("No $include directives found in config.txt! Loading legacy [legacy_configs.Join("/")] files...")
+				log_config("No $include directives found in [DEFAULT_CONFIGURATION_FILE]! Loading legacy [legacy_configs.Join("/")] files...")
```
Comment (Jared-Fogle) [inline code/controllers/configuration/configuration.dm:50]:
```suggestion
				log_config("No [CONFIGURATION_INCLUDE_TOKEN] directives found in [DEFAULT_CONFIGURATION_FILE]! Loading legacy [legacy_configs.Join("/")] files...")
```

File: code/controllers/configuration/configuration.dm [hunk 2]
```diff
@@ -140,15 +149,44 @@
...
+
+/datum/controller/configuration/proc/LoadEntries(filename, list/stack = list())
+	if(IsAdminAdvancedProcCall())
+		return
+
+	var/filename_to_test = world.system_type == MS_WINDOWS ? lowertext(filename) : filename
+	if(filename_to_test in stack)
+		log_config("Warning: Config recursion detected ([english_list(stack)]), breaking!")
+		return
+	stack = stack + filename_to_test
```
Comment (Jared-Fogle) [inline code/controllers/configuration/configuration.dm:169]:
```suggestion
	stack += filename_to_test
```

File: code/modules/unit_tests/configuration_documentation.dm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	var/list/extraneous_entries
+	/// List of entries seen already
+	var/list/seen_entries
+
+/datum/unit_test/configuration_documentation/Run()
+	undocumented_entries = list()
+	for(var/I in global.config.entries)
+		var/datum/config_entry/CE = global.config.entries[I]
+		if(!CE.deprecated_by)
+			undocumented_entries += I
```
Comment (Jared-Fogle) [inline code/modules/unit_tests/configuration_documentation.dm:14]:
```suggestion
	for(var/entry_key in global.config.entries)
		var/datum/config_entry/config_entry = global.config.entries[entry_key]
		if(!config_entry.deprecated_by)
			undocumented_entries += entry_key
```

File: code/modules/unit_tests/configuration_documentation.dm [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	extraneous_entries = list()
+	seen_entries = list()
+	var/test_config_file = world.params["original_config"] ? world.params["original_config"] : DEFAULT_CONFIGURATION_FILE
+	TestGraph(test_config_file)
+	if(undocumented_entries.len)
+		Fail("The following configuration entries are missing default values in the .txt (commented out or otherwise): [english_list(undocumented_entries)]")
+	if(extraneous_entries.len)
```
Comment (Jared-Fogle) [inline code/modules/unit_tests/configuration_documentation.dm:22]: Nothing touches `extraneous_entries`, are you sure this is correct?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/configuration_documentation.dm (hunk 4 above)
Comment (Cyberboss) [reply code/modules/unit_tests/configuration_documentation.dm:22]: It's populated in `TryRemoveEntry` if it finds one it doesn't recognize

**Comment by optimumtact** [comment]
god bless you

File: code/modules/unit_tests/configuration_documentation.dm (hunk 4 above)
Comment (Jared-Fogle) [reply code/modules/unit_tests/configuration_documentation.dm:22]: Didn't realize it was a variable on the test :+1:

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**